sencha-workspace/SlateTasksTeacher/app/view/TaskEditor.js

```diff
@@ -0,0 +1,99 @@
+/* global Slate*/
+Ext.define('SlateTasksTeacher.view.TaskEditor', {
+    extend: 'Slate.cbl.view.modals.CreateTask',
+    xtype: 'slate-tasks-teacher-taskeditor',
+
+    config: {
+        task: null,
+        studentTask: null
+    },
+
+    updateStudentTask: function(studentTask) {
+        var me = this,
+            assignmentsfield = me.down('slate-tasks-assignmentsfield'),
+            assignmentsfieldCheckbox = assignmentsfield.down('checkboxfield'),
+            titleField = me.down('slate-tasks-titlefield'),
+            parentTaskField = titleField.next('slate-tasks-titlefield'),
+            experienceField = me.down('#experience-type'),
+            duedateField = me.down('#duedate'),
+            expirationdateField = me.down('#expirationdate'),
+            skillsField = me.down('slate-skillsfield'),
+            attachmentsfield = me.down('slate-tasks-attachmentsfield'),
+            instructionsField = me.down('#instructions'),
+            taskPrivacyField = me.down('#status');
```

Comment:
@nbey consider providing a `getForm()` method on the modal, grabbing that once, and then using form`s `getField` api for this to grab form field based on their name as intended rather than with itemIds